facilitator/src/transport.rs

```diff
@@ -88,11 +100,314 @@ impl Transport for NullTransport {
     }
 }
 
-// TODO: S3Transport; https://github.com/rusoto/rusoto
+/// Implementation of Transport that reads and writes objects from Amazon S3.
+/// Credentials for authenticating to AWS are automatically sourced from
+/// environment variables or ~/.aws/credentials.
+/// https://github.com/rusoto/rusoto/blob/master/AWS-CREDENTIALS.md
```

Comment:
I punted on more sophisticated credential handling for now because I think we don't yet know how to get our AWS credential to our jobs running in GCP. [There are several ways to provide credentials in Rusoto](https://docs.rs/rusoto_credential/0.45.0/rusoto_credential/index.html).